sdk/iot/Azure.Iot.Hub.Service/src/API Design.md

```diff
@@ -42,7 +42,129 @@ APIs for getting statistics about devices and modules, as well as service statis
+    /// <param name="cancellationToken">The cancellation token</param>
+    /// <returns>The device's twin, including reported properties and desired properties.</returns>
+    public virtual async Task<Response<TwinData>> GetTwinAsync(string deviceId, CancellationToken cancellationToken = default)
+
+    /// <summary>
+    /// Update a device's twin.
+    /// </summary>
+    /// <param name="deviceId">The unique identifier of the device to update the twin on.</param>
+    /// <param name="twinPatch">The properties to update. Any existing properties not referenced by this patch will be unaffected by this patch.</param>
+    /// <param name="ifMatch">A string representing a weak ETag for this twin, as per RFC7232. The update operation is performed
+    /// only if this ETag matches the value maintained by the server, indicating that the twin has not been modified since it was last retrieved.
+    /// To force an unconditional update, set If-Match to the wildcard character (*).</param>
+    /// <param name="cancellationToken">The cancellation token</param>
+    /// <returns>The server's new representation of the device twin.</returns>
+    public virtual async Task<Response<TwinData>> UpdateTwinAsync(TwinData twinPatch, string ifMatch = null, CancellationToken cancellationToken = default)
+
+    /// <summary>
+    /// Update multiple devices' twins. A maximum of 100 updates can be done per call, and each operation must be done on a different device twin. For larger scale operations, consider using IoT Hub jobs (https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-identity-registry#import-and-export-device-identities).
+    /// </summary>
+    /// <param name="twinUpdates">The new twins to replace the twins on existing devices</param>
+    /// <param name="force">If true, all the update operations will ignore the provided twin ETags and will
+    /// force the update. If false, each update operation will fail if the provided ETag for the update is out of date.</param>
+    /// <param name="cancellationToken">The cancellation token</param>
+    /// <returns>The result of the bulk operation.</returns>
+    public virtual async Task<Response<BulkRegistryOperationResult>> UpdateTwinsAsync(IEnumerable<TwinData> twinUpdates, bool force, CancellationToken cancellationToken = default)
+
+    /// <summary>
+    /// Invoke a method on a device.
+    /// </summary>
+    /// <param name="deviceId">The unique identifier of the device to invoke the method on.</param>
+    /// <param name="directMethodRequest">The details of the method to invoke.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>The result of the method invocation.</returns>
+    public virtual async Task<Response<CloudToDeviceMethodResult>> InvokeMethodAsync(string deviceId, CloudToDeviceMethod directMethodRequest, CancellationToken cancellationToken = default)
```

Comment:
Is this object "CloudToDeviceMethodResult" used in jobs? I rem so. Can you double check ? if so do we have issues with separating namespace?

Reply:
CloudToDeviceMethod is used for JobProperties and in this namespace, yes. It is a generated type, though, so it doesn't live in the actual Azure.Iot.Hub.Service.Devices namespace. It lives in Azure.Iot.Hub.Service like all the other generated types. As such, I don't see an issue using it in both the .Jobs and .Devices namespaces